pingpong/ud_pingpong.c

```diff
@@ -479,8 +479,12 @@ static int server_connect(void)
 
 err:
 	free_ep_res();
-	fi_close(&dom->fid);
-	fi_close(&fab->fid);
+	if (dom)
+		fi_close(&dom->fid);
+
+	if (fab)
+		fi_close(&fab->fid);
+
```

Comment:
We're jumping to an error path here.  We should not need if checks.  Just jump to the correct cleanup location.